### ISSUE TITLE
why  the loss in paper  is different from loss in code of the itemrank?

### ISSUE THREAD

**Issue by yuzaitaiyangfei** [cheungdaven/metricfactorization#4]
the loss in paper :
L(P, Q) = (u,i)∈Rcui(Yui−  Pu− Qi 22)  2s.t.   Pu 2≤ l,   Qi 2≤ l
but the loss in code:
self.loss = tf.reduce_sum((1 + 0.1* self.y)*  tf.square((self.y * self.pred + (1 - self.y)  * tf.nn.relu(self.beta * (1 - self.y) - self.pred))))

why they are different?

**Comment by cheungdaven** [comment]
oh, both losses perform equally, I implemented a new contrasive loss here. Anw, the loss in the paper should be easy to implement though.

**Comment by yuzaitaiyangfei** [comment]
Thanks a lot for your kind help!
